### PR TITLE
Fix Sprockets test

### PR DESCRIPTION
**Pull request by tagliala**
**Note to maintainers:** I think this is the expected behavior on Sprockets 4 rather than a bug.

I've tested with `scss` files:

```
# Sprockets 3: 
[1] pry(main)> Sprockets::VERSION
=> "3.7.2"
[2] pry(main)> Rails.application.assets['test.scss'].to_s
=> "/* line 2, app/assets/stylesheets/test.scss */\n.test .nested {\n  color: #fff;\n}\n"


# Sprockets 4:
[1] pry(main)> Sprockets::VERSION
=> "4.0.2"
[2] pry(main)> Rails.application.assets['test.scss'].to_s
=> ".test {\n  .nested {\n    color: #fff;\n  }\n}\n"
[3] pry(main)> Rails.application.assets['test.css'].to_s
=> "/* line 2, app/assets/stylesheets/test.scss */\n.test .nested {\n  color: #fff;\n}\n"
```

Fix Sprockets test

Sprockets 4 require a manifest.js file to be present

Also, different versions of Sprockets return different content according
to file extension:

- Sprockets 2: `.slim` returns HTML
- Sprockets 3: both `.slim` and `.html` return HTML
- Sprockets 4: `.html` returns HTML, `.slim` returns untransformed Slim

### CODE REVIEW

**Comment by tagliala** [comment]
Failure https://travis-ci.org/github/slim-template/slim-rails/jobs/739546076 depends on Rails: rails/rails@892eab777c418135ce0646e91bc9ebb08a29ab9b

Please consider merging this, it does not change production code. A second PR will follow to improve Travis matrix and to address the aforementioned failure, so slim-rails will eventually have a green build :)

**Comment by tagliala** [comment]
Hi @januszm, apologies for the mention

Is there any chance to merge this PR? It does not change production code and does not require a new release.

I would like to submit another one to improve the test matrix but this is required to make all specs green

File: test/lib/slim-rails_assets_test.rb [hunk 1]
```diff
@@ -28,7 +35,8 @@ class DummyApp < ::Rails::Application
 
   test 'compile slim view' do
     assert_equal 'ok', with_app(false, 'print DummyApp.assets || "ok"')
-    assert_equal '<div class="test">hi</div>', with_app(true, 'print DummyApp.assets["test.slim"].to_s')
+    assert_equal '<div class="test">hi</div>', with_app(true, 'print DummyApp.assets["test.slim"].to_s') if Sprockets::VERSION.to_i < 4
+    assert_equal '<div class="test">hi</div>', with_app(true, 'print DummyApp.assets["test.html"].to_s') if Sprockets::VERSION.to_i > 2
```
Comment (januszm) [inline test/lib/slim-rails_assets_test.rb:39]: Thanks for the PR. This part is a bit confusing to me, we have 2 lines of code, one for version < 4 (so e.g. 2 and 3) and second line for version > 2 (so 3 and 4). Both lines should be executed for v3 ?

Comment (tagliala) [reply test/lib/slim-rails_assets_test.rb:39]: Correct. I was thinking to split this test

1. Base case without sprockets (first line)
2. Sprockets 2 responds with compiled template only with `slim` format (second line)
3. Sprockets 3 responds with compiled template with both `html` and `slim`formats (second and third line)
4. Sprockets 4 responds with compiled template with `html` format and with original template with `slim` format (second line + another test to get the original template)

What do you think?

Comment (tagliala) [reply test/lib/slim-rails_assets_test.rb]: Please check if you like the new approach

**Comment by januszm** [comment]
@tagliala merged, thanks and apologies for late reply! Also I think we should add both sprockets 3 and 4 in appraisal gemfiles

**Pull request merged**